[ci] Add Linux Arm64 for release-automation pipeline

khluu opened the pull request:
## Changes
- Rename `forge.Dockerfile` and `forge.wanda.yaml` to be `forge_x86_864`. If you prefer it to just be `forge` and arm64 files to be `forge.arm64/forge.aarch64` to be consistent with other files, I can do that too.
- Add wanda & Dockerfile for Linux Arm64.

## Checks
https://buildkite.com/ray-project/release-automation/builds/22 passes for Python sanity check on both x86_64 and Arm64

khluu commented:
@aslonnie  I followed other files in `ci/docker` for clang installation. Is it the correct way to do it for this case? I remember you saying that you have `clang 14` on your machine but I think OSS CI still uses `clang 12`.

aslonnie reviewed (commented):
can we split the c++ sanity check and arm64 into 2 separate PR's?

khluu commented:
> can we split the c++ sanity check and arm64 into 2 separate PR's?

I reverted this PR into just adding Arm64 and its Python sanity check. Bazel, Clang installation and C++ sanity check are moved to #43094

aslonnie commented on .buildkite/release-automation/wheels.rayci.yml:
```diff
@@ -1,10 +1,28 @@
+    wanda: .buildkite/release-automation/forge_arm64.wanda.yaml
+    instance_type: builder-arm64
+
+  - label: "Linux Arm64 Python {{matrix}}"
```
```suggestion
  - label: "Linux arm64 Python {{matrix}}"
```

Either `ARM64` or `arm64`
